source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift

```diff
@@ -226,18 +231,58 @@ class ACRContentStackView: NSView, ACRContentHoldingViewProtocol, SelectActionHa
         layer?.backgroundColor = ColorUtils.hoverColorOnMouseEnter().cgColor
     }
     
-    func setErrorMessage(with title: String, for view: InputHandlingViewProtocol) {
-        guard errorMessageView == nil else { return }
+    private func staticTextField() -> NSTextField {
         let textField = NSTextField()
-        textField.stringValue = title
-        textField.isHidden = true
+        textField.allowsEditingTextAttributes = true
         textField.isEditable = false
         textField.isBordered = false
         textField.isSelectable = true
         textField.backgroundColor = .clear
+        return textField
+    }
+    
+    func configureInputElements(element: ACSBaseInputElement, view: NSView) {
+        setupLabel(for: element)
+        addArrangedSubview(view)
+        if view is ACRContentStackView {
+            view.widthAnchor.constraint(equalTo: widthAnchor).isActive = true
+        }
+        setupErrorMessage(element: element, view: view)
+    }
+    
+    private func setupLabel(for element: ACSBaseInputElement) {
+        guard renderConfig.supportsSchemeV1_3, inputLabelField?.stringValue != element.getLabel() else { return }
+        if let label = element.getLabel(), !label.isEmpty {
+            let attributedString = NSMutableAttributedString(string: label)
+            let errorStateConfig = renderConfig.inputFieldConfig.errorStateConfig
+            let isRequiredSuffix = (hostConfig.getInputs()?.label.requiredInputs.suffix ?? "").isEmpty ? "*" : hostConfig.getInputs()?.label.requiredInputs.suffix ?? "*"
+            if let colorHex = hostConfig.getForegroundColor(style, color: .default, isSubtle: false), let textColor = ColorUtils.color(from: colorHex) {
+                attributedString.addAttributes([.foregroundColor: textColor, .font: NSFont.systemFont(ofSize: 16)], range: NSRange(location: 0, length: attributedString.length))
+            }
+            if element.getIsRequired() {
+                attributedString.append(NSAttributedString(string: " " + isRequiredSuffix, attributes: [.foregroundColor: errorStateConfig.textColor, .font: NSFont.systemFont(ofSize: 16)]))
+            }
+            let labelView = staticTextField()
+            labelView.attributedStringValue = attributedString
+            addArrangedSubview(labelView)
+            setCustomSpacing(spacing: 3, after: labelView)
+            inputLabelField = labelView
+        }
+    }
+    
+    func setupErrorMessage(element: ACSBaseInputElement, view: NSView) {
+        guard renderConfig.supportsSchemeV1_3, errorMessageField?.stringValue != element.getErrorMessage() else { return }
```

Comment:
```suggestion
    private func setupErrorMessage(element: ACSBaseInputElement, view: NSView) {
        guard renderConfig.supportsSchemeV1_3, errorMessageField?.stringValue != element.getErrorMessage() else { return }
```
Maybe can be private now